lib/segment/src/data_types/named_vectors.rs

```diff
@@ -14,11 +16,17 @@ use crate::types::{Distance, VectorDataConfig, VectorStorageDatatype};
 
 type CowKey<'a> = Cow<'a, str>;
 
+#[derive(Clone, PartialEq, Debug)]
+pub enum CowMultiVector<'a, TElement: PrimitiveVectorElement> {
```

Comment:
We cannot use regular `Cow<MultiDenseVector>` because we don't use `&MultiDenseVector`. Introduce custom enum with owned and borrowed states